src/octoprint/timelapse.py

```diff
@@ -838,7 +839,40 @@ def _render(self):
 		with self.render_job_lock:
 			try:
 				self._notify_callback("start", output)
-				p = sarge.run(command_str, stdout=sarge.Capture(), stderr=sarge.Capture())
+				p = sarge.run(command_str, stdout=sarge.Capture(), stderr=sarge.Capture(), async=True)
+
+				self._logger.debug("Parsing ffmpeg output")
+
+				# Compile regex to make it faster
+				duration_regex = re.compile("Duration: (\d{2}):(\d{2}):(\d{2})\.\d{2}")
+				current_regex = re.compile("time=(\d{2}):(\d{2}):(\d{2})\.\d{2}")
+				duration_s = ""
+
+				# Sarge sometimes doesn't populate commands fast enough, so we check if the command exists first, then poll the thread
+				while len(p.commands) == 0 or p.commands[0].poll() is None:
+					line = p.stderr.readline()
+
+					# We should be checking time more often, so try it first
+					time = current_regex.search(line)
+					if time is not None:
+						current_s = self._parsetime(*time.groups())
+
+						progress = current_s / float(duration_s) * 100
```

Comment:
We might not have had a duration yet, so it still could be zero, which would lead to a division by zero here. Needs some checking to make sure this can never happen.